### ISSUE TITLE
[java] UseArrayListInsteadOfVector ignores Vector when other classes are imported

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#4148]
**Affects PMD Version:6.47.0** 

**Rule:UseArrayListInsteadOfVector**

Please provide the rule name and a link to the rule documentation:
<https://pmd.sourceforge.io/pmd-6.50.0/pmd_rules_java_performance.html#usearraylistinsteadofvector>

**Description:**
Hi, I found a false negative about the rule UseArrayListInsteadOfVector, please read the first code example below. PMD should have reported a warning in line 5, but it does not. However, in the second code example, PMD can report the warning as expected. Hence, I think this is a false negative. Thank you!

**Code Sample demonstrating the issue:**

**First code example**
```java
import org.Annotation;
@Annotation
public class C {
    void x() {
        Vector v = new Vector();  // should report a warning in this line, but no warnings
    }
}
```

**Second code example**
```java
public class C {
    void x() {
        Vector v = new Vector();  // Can detect the warning
    }
}
```

**Expected outcome:**

PMD should report a violation at line 5, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Belle-PL** [comment]
Hi, thanks for your reply. I may ignore the import when submitted the issue, and it seems that problem is not only casuse by the annotation and has been fixed now.
